### PR TITLE
fix: make `dialog_file_browser_set_basic_options` initialize all fields

### PR DESCRIPTION
**Pull request by JarvisCraft**
# What's new

Previously `dialogs_file_browser_options` did not initialize `hide_dot_files` which led to undefined behaviour in most of its use-cases since it was called on an unitialized structs leaving them in uninitialied state.
This PR fixes this by adding initialization for the remaining field.
This also updates the documentation to ensure that `dialogs_file_browser_options` should initialize all fields and adds a unit-test to assert this behaviour.

# Verification 

- Unit tests should be run, i.e. `dialogs_file_browser_options` suite.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Review by skotopes — DISMISSED** [review]

**Comment by JarvisCraft** [comment]
The test seems to fail, I will take a look.

File: applications/debug/unit_tests/dialogs/dialogs_file_browser_options.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#include <dialogs/dialogs.h>
+
+#include "../minunit.h"
+
+MU_TEST(test_dialog_file_browser_set_basic_options_should_init_all_fields) {
```
Comment (skotopes) [inline applications/debug/unit_tests/dialogs/dialogs_file_browser_options.c:5]: This test is incorrect: DialogsFileBrowserOptions structure is not packed and filling it with `n` is not going to work

**Review by skotopes — CHANGES_REQUESTED** [review]

File: applications/debug/unit_tests/dialogs/dialogs_file_browser_options.c (hunk 1 above)
Comment (JarvisCraft) [reply applications/debug/unit_tests/dialogs/dialogs_file_browser_options.c:5]: Hmm, I forgot about alignment. In this case, I am not sure what the right way to test this (allowing automatic updates in case of futue changes) would be.

Comment (JarvisCraft) [reply applications/debug/unit_tests/dialogs/dialogs_file_browser_options.c:5]: I've got a small idea to utilize minimal test to ensure some compatibility.

Comment (JarvisCraft) [reply applications/debug/unit_tests/dialogs/dialogs_file_browser_options.c:5]: @skotopes, here is the alternative approach to the test. It may not handle some specific corner-cases but it is better than nothing.

It adds an assertion on structure size (guess I calculated it correctly) which would get triggered on most changes to `DialogsFileBrowserOptions` (except for ones which fit into alignment) forcing the developer to check the test and thus update initialization logic.

**Comment by JarvisCraft** [comment]
@skotopes, could you please take a look at the changes and re-run the CI?

**Review by skotopes — APPROVED** [review]
